content/en/agent/docker/_index.md

```diff
@@ -122,14 +122,14 @@ Optional collection Agents are disabled by default for security or performance r
 
 Send custom metrics with [the StatsD protocol][14]:
 
-| Env Variable                     | Description                                                                                                                                                |
-|----------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| `DD_DOGSTATSD_NON_LOCAL_TRAFFIC` | Listen to DogStatsD packets from other containers (required to send custom metrics).                                                                       |
-| `DD_HISTOGRAM_PERCENTILES`       | The histogram percentiles to compute (separated by spaces). The default is `0.95`.                                                                         |
-| `DD_HISTOGRAM_AGGREGATES`        | The histogram aggregates to compute (separated by spaces). The default is "max median avg count".                                                          |
-| `DD_DOGSTATSD_SOCKET`            | Path to the unix socket to listen to. Must be in a `rw` mounted volume.                                                                                    |
-| `DD_DOGSTATSD_ORIGIN_DETECTION`  | Enable container detection and tagging for unix socket metrics.                                                                                            |
-| `DD_DOGSTATSD_TAGS`              | Additional tags to append to all metrics, events, and service checks received by this DogStatsD server, for example: `["env:golden", "group:retrievers"]`. |
```

Comment:
Hi @sveniu,

Actually both works. The agent support JSON env vars as well as space separated list for `[]string` type.

I tired with `7.22.0`:
```bash
$> DD_DOGSTATSD_TAGS='["env:golden", "group:retrievers"]' datadog-agent run
$> datadog-agent config
[...]
dogstatsd_tags:
- env:golden
- group:retrievers
[...]

// As well as

$> DD_DOGSTATSD_TAGS="env:golden group:retrievers" datadog-agent run
$> datadog-agent config
[...]
dogstatsd_tags:
- env:golden
- group:retrievers
[...]
```

Could you tell me more about your case where the JSON version didn't worked ?

Reply:
Hi @hush-hush. I tried with 7.21.1.

With JSON encoding:
```
docker run --rm -ti \
-e DD_DOGSTATSD_TAGS='["env:golden", "group:retrievers"]' \
-e DD_API_KEY=1234 \
--name ddagent \
datadog/agent:7.21.1

docker exec ddagent agent config | grep -A2 ^dogstatsd_tags
dogstatsd_tags:
- '["env:golden",'
- '"group:retrievers"]'
```

With spaces:
```
docker run --rm -ti \
-e DD_DOGSTATSD_TAGS='env:golden group:retrievers' \
-e DD_API_KEY=1234 \
--name ddagent \
datadog/agent:7.21.1

docker exec ddagent agent config | grep -A2 ^dogstatsd_tags
dogstatsd_tags:
- '["env:golden",'
- '"group:retrievers"]'
```

It is indeed fixed in 7.22.0, although I didn't dig into where the fix actually is (spf13's viper or cast, most likely).

```
docker run --rm -ti \
-e DD_DOGSTATSD_TAGS='["env:golden", "group:retrievers"]' \
-e DD_API_KEY=1234 \
--name ddagent \
datadog/agent:7.22.0

docker exec ddagent agent config | grep -A2 ^dogstatsd_tags
dogstatsd_tags:
- env:golden
- group:retrievers
```

So it is indeed fixed in 7.22.0. I guess it's up to you whether a doc fix or note is warranted, depending on how quickly most clients would be upgraded to 7.22.0 or newer... I'm fine with closing this, though.